test/test_schedule.py

```diff
@@ -136,6 +136,7 @@ def test_cache_binaryop_reshaped(self):
     d = a.reshape(10,1)+b.reshape(10,1)
     check_schedule(d, 0, [c])
 
+  @unittest.skip("not pushing movement ops through sourceless buffers")
```

Comment:
so the logic is that check till the **movementroot** of **ast.op.buffers** and if all of them are realized/sourceless don't push.

this will make the following tests pass
```
  def test_no_binop_rerun(self): return self._test_no_binop_rerun(lambda a,b: a*b, lambda a,b: (a*b).reshape(16, 16, 1))
  def test_no_binop_rerun_alt(self): return self._test_no_binop_rerun(lambda a,b: (a*b).reshape(16, 16, 1), lambda a,b: a*b)
  def test_no_binop_rerun_reduce_broadcast(self): return self._test_no_binop_rerun(lambda a,b: a.sum()+b, lambda a,b: a.sum().reshape(1,1)+b, allowed=2)
```

but (a.T * b.T).T will fail. We need to decide which is more important at the very bottom of the computation graph. 

btw this test is already disabled "test_no_binop_rerun_transposed" in external_test_opt

```
  @unittest.skip("this test started failing with the new change, based movementop issue")
  def test_no_binop_rerun_transposed(self): return self._test_no_binop_rerun(lambda a,b: (a.T*b.T).T, lambda a,b: a*b)
```

Reply:
Disabling this is okay